mayuctl/list.go

```diff
@@ -2,44 +2,144 @@ package main
 
 import (
 	"fmt"
+	"os"
+	"sort"
+	"strings"
 
 	"github.com/giantswarm/mayu/hostmgr"
 	"github.com/ryanuber/columnize"
 	"github.com/spf13/cobra"
 )
 
+const (
+	defaultListFields = "ip,serial,profile,ipmiaddr,providerid,metadata,coreos,state,lastboot"
+	timestampFormat   = "2006-01-02 15:04:05"
+)
+
+type ListFlags struct {
+	Fields   string
+	NoLegend bool
+}
+
+type hostToField func(host *hostmgr.Host) string
+
 var (
 	listCmd = &cobra.Command{
-		Use:   "list",
+		Use:   "list [--no-legend] [--fields]",
 		Short: "List machines.",
-		Long:  "List machines.",
-		Run:   listRun,
+		Long: `List the state of all machines in a Mayu datacenter.
+
+For easily parsable output, you can remove the column headers:
+	mayuctl list --no-legend
+
+Or, choose the columns to display:
+	mayuctl list --fields=ip,machineid,ipmiaddr`,
+		Run: listRun,
 	}
-)
 
-const (
-	listHeader      = "IP | Serial | Profile | IPMI Address | ProviderId | Fleet | CoreOS | State | Last Boot"
-	listScheme      = "%s | %s | %s | %s | %s | %s | %s | %s | %s"
-	timestampFormat = "2006-01-02 15:04:05"
+	listFields = map[string]hostToField{
```

Comment:
@teemow Why don't you use a generic function with a switch instead of a map that stores sometimes similar functions ? I don't see the benefits of using this map instead of a function with a switch.